src/web_analysis/test_robots.py

```diff
@@ -0,0 +1,1087 @@
+        agent_rules = {"*": {"Disallow": ["/folder"], "Allow": ["/folder/subfolder"]}}
+        all_agents = ["GoogleBot"]
+        expected = {"*": "some", "GoogleBot": "some"}
+        self.assertEqual(
+            interpret_robots(agent_rules, all_agents),
+            expected,
+            "Allow should take precedence over Disallow when both target the same path hierarchy.",
+        )
+
+    def test_wildcard_handling(self):
+        """Tests the correct interpretation of wildcard characters in paths."""
+        agent_rules = {"*": {"Disallow": ["/tmp/*"]}}
+        all_agents = ["GoogleBot"]
+        expected = {"*": "some", "GoogleBot": "some"}
+        self.assertEqual(
+            interpret_robots(agent_rules, all_agents),
+            expected,
+            "Wildcards should block access to all subdirectories under /tmp, but they do not.",
+        )
+
+    def test_mixed_case_user_agents(self):
+        """Tests whether the function correctly handles mixed-case user-agent names."""
+        agent_rules = {
+            "Googlebot": {"Disallow": ["/private"]},
+            "googleBot": {"Allow": ["/private/stats"]},
+        }
+        all_agents = ["Googlebot"]
+        expected = {"*": "none", "Googlebot": "some"}
+        self.assertEqual(
+            interpret_robots(agent_rules, all_agents),
+            expected,
+            "User-agent names should be treated case-insensitively, but they are not.",
+        )
+
+    def test_query_string_sensitive_blocking(self):
+        """Tests handling of URLs that include query strings in their disallow rules."""
+        agent_rules = {"*": {"Disallow": ["/search?query=*"]}}
+        all_agents = ["GoogleBot"]
+        expected = {"*": "some", "GoogleBot": "some"}
+        self.assertEqual(
+            interpret_robots(agent_rules, all_agents),
+            expected,
+            "Disallowed paths with query strings should block matching URLs, but they do not.",
+        )
+
+    def test_empty_robots_txt(self):
+        """Tests behavior when the robots.txt is empty, implying all access should be allowed."""
+        agent_rules = {}
+        all_agents = ["GoogleBot"]
+        expected = {"*": "none", "GoogleBot": "none"}
+        self.assertEqual(
+            interpret_robots(agent_rules, all_agents),
+            expected,
+            "An empty robots.txt should allow all access, but it does not.",
+        )
+
+    def test_precedence_of_directives(self):
+        """Tests precedence of Allow over Disallow directives within the same robots.txt."""
+        agent_rules = {"GoogleBot": {"Disallow": ["/"], "Allow": ["/public"]}}
+        all_agents = ["GoogleBot"]
+        expected = {"*": "none", "GoogleBot": "some"}
+        self.assertEqual(
+            interpret_robots(agent_rules, all_agents),
+            expected,
+            "Allow directives should override conflicting Disallow directives, but they do not.",
+        )
+
+    def test_disallow_all_but_one(self):
```

Comment:
This same test appears twice?

Reply:
Also, are we sure that * would be disallowed from all? I think Google standard would give it `some` as well?

Reply:
Fixed this issue. Seems like the index.html is an exception so it should be some according to the Google standards.